ext/spl/spl_iterators.stub.php

```diff
@@ -0,0 +1,475 @@
+<?php
+
+class EmptyIterator implements Iterator
+{
+    /** @return void */
+    public function current() {}
+
+    /** @return void */
+    public function next() {}
+
+    /** @return void */
+    public function key() {}
+
+    /** @return bool */
+    public function valid() {}
+
+    /** @return void */
+    public function rewind() {}
+}
+
+class CallbackFilterIterator extends FilterIterator
+{
+    public function __construct(Iterator $iterator, callable $callback) {}
+
+    /** @return bool */
+    public function accept() {}
+}
+
+class RecursiveCallbackFilterIterator extends CallbackFilterIterator implements RecursiveIterator
+{
+    public function __construct(RecursiveIterator $iterator, callable $callback) {}
+
+    /** @return bool */
+    public function hasChildren() {}
+
+    /** @return RecursiveCallbackFilterIterator */
+    public function getChildren() {}
+}
+
+interface RecursiveIterator extends Iterator
+{
+    /** @return bool */
+    public function hasChildren();
+
+    /** @return RecursiveIterator */
+    public function getChildren();
+}
+
+class RecursiveIteratorIterator implements OuterIterator
+{
+    /** @param Traversable $iterator */
+    public function __construct(Traversable $iterator, int $mode = self::LEAVES_ONLY, int $flags = 0) {}
+
+    /** @return void */
+    public function rewind() {}
+
+    /** @return bool */
+    public function valid() {}
+
+    /** @return mixed */
+    public function key() {}
+
+    /** @return mixed */
+    public function current() {}
+
+    /** @return void */
+    public function next() {}
+
+    /** @return int */
+    public function getDepth() {}
+
+    /** @return RecursiveIterator */
+    public function getSubIterator(int $level = UNKNOWN) {}
+
+    /** @return RecursiveIterator */
+    public function getInnerIterator() {}
+
+    /** @return void */
+    public function beginIteration() {}
+
+    /** @return void */
+    public function endIteration() {}
+
+    /** @return bool|null */
```

Comment:
We might want to change that one null condition to false. Doesn't make sense to return null on a boolean method.

Most likely it's just dead code, as callGetChildren() does not contain a check for object->iterators...